lib/form/html.js

```diff
@@ -29,7 +29,46 @@ const addError = (html, input, errors) => {
     });
   }
 
-  result += html;
+  result += render;
+
+  return result;
+};
+
+/*
+ * Add Required Rendered Inputs
+ *
+ * @param {string} html - Rendered HTML
+ * @param {InputType} type - Input type being rendered
+ *
+ * @returns {string} - Rendered HTML, with required indicators
+ */
+const addRequired = (html, input) => {
+  let render = html;
+  let result = '';
+
+  Object.keys(input.inputs).map(inp => {
+    if (((input.hasOwnProperty('required')) && input.required === 'save') || ((input.inputs[inp].hasOwnProperty('required')) && input.inputs[inp].required === 'save')) {
+      const id = input.inputs[inp].id;
+      let level = 'required ';
+      if (input.inputs[inp].hasOwnProperty('required')) {
+        level += `required--${input.inputs[inp].required}`;
+      }
+      else {
+        level += `required--${input.required}`;
+      }
+
+      // add required class to label
+      let find = new RegExp(`for=['"]\\s*${id}\\s*['"]`);
+      render = render.replace(find, `for="${id}" class="${level}"`);
+
+      // add required to input
+      const name = `${input.id}--${inp}`;
+      find = new RegExp(`name=['"]\\s*${name}\\s*['"]`);
+      render = render.replace(find, `name="${name}" aria-required="true" required`);
```

Comment:
Hmm, we should talk about whether we should add the `required` attribute if it's required to publish